ssh-agent-canonicalize

```diff
@@ -1,10 +1,19 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+# Get userid on MacOS
+if [[ $OSTYPE == 'darwin'* ]]; then
```

Comment:
The BSD options should work fine for the GNU versions of these commands. Let's just simplify it to use the short-opts. Then we don't need any variables or anything; we can just update the use of the args below.